### ISSUE TITLE
Support more than one parameter groups

### ISSUE THREAD

**Issue by phamnam95** [jotaf98/pytorch-curveball#1]
Hello. Thank you for publishing the code. Are there any ways to change the code so that it supports more than one parameter groups? Thank you.

**Comment by jotaf98** [comment]
Yes, I only implemented it like this as a simplification, since groups are redundant when the hyper-parameters are set automatically.

So a use case for groups -- setting hyper-parameters separately per group (e.g. learning rate) -- disappears, since there's no need to set the learning rate.

I realize now that another use case is setting different weight decays, which is reasonable.

Anyway, the easiest way is to just flatten the groups into a single group, which will be equivalent to a proper CurveBall implementation supporting groups (except for the ability to set individual weight decays).
